python/cproton.py

```diff
@@ -204,12 +204,12 @@ def bytes2string(b, encoding='utf8'):
     return ffi.unpack(b.start, b.size).decode(encoding)
 
 
-def py2bytes(py):
+def py2bytes(py, encoding='utf8'):
     if isinstance(py, (bytes, bytearray, memoryview)):
         s = ffi.from_buffer(py)
         return len(s), s
     elif isinstance(py, str):
-        s = ffi.from_buffer(py.encode('utf8'))
+        s = ffi.from_buffer(py.encode(encoding))
         return len(s), s
```

Comment:
Not entirely sure about this change - as it's only necessary because of the change below for round tripping.